src/js/components/charts/StackedProgressBar.js

```diff
@@ -0,0 +1,90 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import ServicePlanStatusTypes from '../../constants/ServicePlanStatusTypes';
+
+class StackedProgressBar extends React.Component {
+  render() {
+    let label;
+    let secondaryLabel;
+    let props = this.props;
+    let classes = classNames(props.className, props.layoutClassName, {
+      'is-complete': props.progressState === ServicePlanStatusTypes.COMPLETE,
+      'is-erroneous': props.progressState === ServicePlanStatusTypes.ERROR,
+      'is-ongoing': props.progressState === ServicePlanStatusTypes.IN_PROGRESS,
+      'is-upcoming': props.progressState === ServicePlanStatusTypes.PENDING,
+      'is-waiting': props.progressState === ServicePlanStatusTypes.WAITING
+    });
+
+    if (props.label) {
+      label = (
+        <div className={props.headerClassName}>
+          <span className={props.labelClassName}>
+            <span className={props.labelContentClassName}>{props.label}</span>
+          </span>
+          <span className={props.labelActionClassName}>
+            {props.labelAction}
+          </span>
+        </div>
+      );
+    }
+
+    if (props.secondaryLabel) {
+      secondaryLabel = (
+        <div className={props.secondaryLabelClassName}>
+          {this.props.secondaryLabel}
+        </div>
+      );
+    }
+
+    return (
+      <div className={classes}>
+        {label}
+        <div className={props.fillWrapperClassName}>
+          <div className={props.fillClassName} style={{
+            width: `${props.progress}%`
+          }} />
+        </div>
+        {secondaryLabel}
+      </div>
+    );
+  }
+}
+
+StackedProgressBar.defaultProps = {
+  className: 'stacked-progress-bar progress-bar',
```

Comment:
i was going to say, let's go with the new way of doing default classnames, but since these are almost all one classers, this is fine